src/main/java/com/mathworks/ci/RunMatlabTestsBuilder.java

```diff
@@ -149,15 +149,27 @@ private EnvVars getEnv() {
     
     // To retain Backward compatibility
     protected Object readResolve() {
-        // Assign default values to new elements.
-        this.pdfReportArtifact = new NullArtifact();
-        this.tapArtifact = new NullArtifact();
-        this.junitArtifact = new NullArtifact();
-        this.coberturaArtifact = new NullArtifact();
-        this.stmResultsArtifact = new NullArtifact();
-        this.modelCoverageArtifact = new NullArtifact();
-
-        // Assign appropriate artifact type if it was selected earlier.
+        // Assign default values to new elements if not already deserialized from config.
+        if (this.pdfReportArtifact == null) {
+            this.pdfReportArtifact = new NullArtifact();
+        }
+        if (this.tapArtifact == null) {
+            this.tapArtifact = new NullArtifact();
+        }
+        if (this.junitArtifact == null) {
+            this.junitArtifact = new NullArtifact();
+        }
+        if (this.coberturaArtifact == null) {
+            this.coberturaArtifact = new NullArtifact();
+        }
+        if (this.stmResultsArtifact == null) {
+            this.stmResultsArtifact = new NullArtifact();
+        }
+        if (this.modelCoverageArtifact == null) {
+            this.modelCoverageArtifact = new NullArtifact();

```

Comment:
Changes looks good . You may want to consider using Java Optional instead of null check  it will also avoid the multiple if loops. 
eg: 
 Optional.ofNullable(this.pdfReportArtifact).orElseGet(() -> new NullArtifact());

Reply:
Awesome! applied these changes, thanks!